src/main/java/com/yevhenberladyniuk/debts/service/DebtServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+package com.yevhenberladyniuk.debts.service;
+
+import com.yevhenberladyniuk.debts.domain.Debt;
+import com.yevhenberladyniuk.debts.domain.Partner;
+import com.yevhenberladyniuk.debts.domain.User;
+import com.yevhenberladyniuk.debts.dto.CreateDebtForm;
+import com.yevhenberladyniuk.debts.dto.DebtDto;
+import com.yevhenberladyniuk.debts.repository.DebtRepository;
+import com.yevhenberladyniuk.debts.repository.PartnerRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDateTime;
+import java.util.List;
+
+@Service
+public class DebtServiceImpl implements DebtService{
+
+    private DebtRepository debtRepository;
+
+    private PartnerRepository partnerRepository;
+
+    private  PartnerService partnerService;
+
+    @Autowired
+    public DebtServiceImpl(DebtRepository debtRepository, PartnerRepository partnerRepository, PartnerService partnerService) {
+        this.debtRepository = debtRepository;
+        this.partnerRepository = partnerRepository;
+        this.partnerService = partnerService;
+    }
+
+    @Override
+    public List<Debt> findAllByPartnerId(Long partnerId, User user) {
+        checkAccessToPartner(partnerId, user);
+        return debtRepository.findAllByPartnerIdOrderByTransactionDateDesc(partnerId);
+    }
+
+    @Override
+    @Transactional
+    public void create(CreateDebtForm createDebt, User user) {
+
+        LocalDateTime transactionDate = createDebt.getTransactionDate() == null ?
+                                                            LocalDateTime.now() : createDebt.getTransactionDate() ;
+
+        Debt debt = Debt.builder()
+                .partnerId(createDebt.getPartnerId())
+                .comment(createDebt.getComment())
+                .transactionAmount(createDebt.getTransactionAmount())
+                .transactionDate(transactionDate)
+                .build();
+
+        debtRepository.save(debt);
+
+        Partner partner = checkAccessToPartner(createDebt.getPartnerId(), user);
```

Comment:
This line should be at the beginning of the method.
Cause you firstly save new debt, and then check access :D

Reply:
Oops)